source/Calamari.Aws/Deployment/Conventions/UploadAwsS3Convention.cs

```diff
@@ -140,11 +140,43 @@ private void SetOutputVariables(RunningDeployment deployment, IEnumerable<S3Uplo
         {
             log.SetOutputVariableButDoNotAddToVariables(PackageVariables.Output.FileName, Path.GetFileName(deployment.PackageFilePath));
             log.SetOutputVariableButDoNotAddToVariables(PackageVariables.Output.FilePath, deployment.PackageFilePath);
-            foreach (var result in results)
+            var successfulResults = results.Where(z => z.IsSuccess()).ToArray();
+            var actionName = deployment.Variables["Octopus.Action.Name"];
+            if (successfulResults.Length == 1)
```

Comment:
Should this be based on whether they have selected "upload whole package" or not as opposed to the number of files successfully uploaded? If they've selected to "upload some files" they won't necessarily know at configuration time how many files will match the glob pattern (it could be 1) and so won't know which output variables to use.